Source/UI/IdentityUI.swift

```diff
@@ -477,6 +491,115 @@ extension IdentityUI {
+                }
+
+                func success(user: User) {
+                    completion(.success(user: user, persistUser: true))
+                }
+
+                func cancel() {
+                    completion(.cancel)
+                }
+
+                func presentTerms(user: User) {
+                    self._identityManager.fetchTerms { result in
+                        switch result {
+                        case .success(let terms):
+                            let navigationController = DismissableNavigationController {
+                                if IdentityUI.updatedTermsCoordinator != nil {
+                                    user.logout()
+                                    cancel()
+                                }
+                                IdentityUI.updatedTermsCoordinator = nil
+                            }
+
+                            let updatedTermsCoordinator = UpdatedTermsCoordinator(
+                                navigationController: navigationController, configuration: self.configuration)
+
+                            updatedTermsCoordinator.start(input: .init(currentUser: user, terms: terms)) { output in
+                                IdentityUI.updatedTermsCoordinator = nil
+
+                                navigationController.dismiss(animated: true) {
+                                    switch output {
+                                    case .success:
+                                        success(user: user)
+                                    case .cancel:
+                                        cancel()
+                                    }
+                                }
+                            }
+
+                            IdentityUI.updatedTermsCoordinator = updatedTermsCoordinator
+                            self.configuration.presentationHook?(navigationController)
+                            presentingViewController.present(navigationController, animated: true, completion: nil)
+                        case .failure(let error):
+                            log(from: self, error)
+                            user.logout()
+                            self.present(error: error)
+                        }
+                    }
+                }
+
+                self._identityManager.login(username: .email(email),
+                                            password: password,
+                                            scopes: scopes,
+                                            persistUser: true,
+                                            useSharedWebCredentials: false) { result in
+                    switch result {
+                    case .failure(let error):
+                        log(from: self, error)
+                        fallback(.passwordWithPrefilledEmail(email))
+                    case .success:
+                        let user = self._identityManager.currentUser
+                        user.agreements.status { result in
```

Comment:
Instead of this re-implementation of asking user to accept agreements, I think `CompleteProfileCoordinator` could be re-used.

Something like:
```swift
let coordinator = CompleteProfileCoordinator(navigationController: navigationController,
                                             identityManager: self._identityManager,
                                             configuration: self.configuration)
let profileCoordinator = UpdateProfileInteractor(currentUser: user,
                                                 loginFlowVariant: .signin,
                                                 tracker: self.configuration.tracker)
```

Reply:
The problem is we can't present the other coordinators from this flow, as they don't have a completion block. This results in the consumer of the API (the app) doesn't know whether the login succeeded or not, and can't update it's UI.

Reply:
And remember, we also want to do this without showing any UI unless necessary. From what I've seen, the coordinators flow _always_ presents something.

Reply:
Ah, CompleteProfileCoordinator is special in that it checks all the paths and doesn't present anything unless it needs to. (That's different from the UpdateProfileCoordinator). That might work.